src/Compilers/CSharp/Test/Symbol/Symbols/SymbolEqualityTests.cs

```diff
@@ -750,51 +727,210 @@ public class B
+            var field2Type = field2.Type; // A<T!>
+
+            VerifyEquality(field1Type, field2Type,
+                expectedIncludeNullability: true
+                );
+
+            var field1ContainingType = field1.ContainingType; //A<T>
+            var field2ContainingType = field2.ContainingType; //A<T!>
+
+            VerifyEquality(field1ContainingType, field2ContainingType,
+                expectedIncludeNullability: false
+                );
+
+        }
+
+        [Fact]
+        [WorkItem(8195, "https://github.com/dotnet/roslyn/issues/38195")]
+        public void SemanticModel_SubstitutedMethod_Equality()
+        {
+            var source =
+@"
+#nullable enable
+public class A<T> 
+    where T : class //not neccesary, but makes it easier to reason about the resulting fields
+{
+    public A<T> M(A<T> t)
+    {
+        t.M(t);
+        return t;
+    }
+}
+";
+            var comp = CreateCompilation(source);
+            comp.VerifyDiagnostics();
+
+            var syntaxTree = comp.SyntaxTrees[0];
+            var root = syntaxTree.GetRoot();
+
+            var member1Syntax = (ClassDeclarationSyntax)root.DescendantNodes().First(sn => sn.Kind() == SyntaxKind.ClassDeclaration);
+            var member2Syntax = (IdentifierNameSyntax)root.DescendantNodes().Last(sn => sn.Kind() == SyntaxKind.SimpleMemberAccessExpression).DescendantNodes().Last(sn => sn.Kind() == SyntaxKind.IdentifierName);
+
+            var model = comp.GetSemanticModel(syntaxTree);
+
+            var method1 = (IMethodSymbol)((INamedTypeSymbol)model.GetDeclaredSymbol(member1Syntax)).GetMembers("M").Single(); // A<T!>! A<T>.M(A<T!>! t)
+            var method2 = (IMethodSymbol)model.GetSymbolInfo(member2Syntax).Symbol;                                           // A<T!>! A<T!>.M(A<T!>! t)
+
+            VerifyEquality(method1, method2,
+                expectedIncludeNullability: false
+                );
+
+            var method1ReturnType = method1.ReturnType; // A<T!>
+            var method2ReturnType = method2.ReturnType; // A<T!>
+
+            VerifyEquality(method1ReturnType, method2ReturnType,
+                expectedIncludeNullability: true
+                );
+
+            var method1ParamType = method1.Parameters.First().Type; // A<T!>
+            var method2ParamType = method2.Parameters.First().Type; // A<T!>
+
+            VerifyEquality(method1ParamType, method2ParamType,
+                expectedIncludeNullability: true
+                );
+
+            var method1ContainingType = method1.ContainingType; //A<T>
+            var method2ContainingType = method2.ContainingType; //A<T!>
+
+            VerifyEquality(method1ContainingType, method2ContainingType,
+                expectedIncludeNullability: false
+                );
+        }
+
+        [Fact]
+        [WorkItem(8195, "https://github.com/dotnet/roslyn/issues/38195")]
+        public void SemanticModel_SubstitutedEvent_Equality()
```

Comment:
>SemanticModel_SubstitutedEvent_Equality [](start = 20, length = 39)

It doesn't look like we made changes for events. Why does this test succeed?

Reply:
EventSymbol already has a custom equality/hashcode implementation and SubsititutedEventSymbol doesn't override it.

(note: my previous PR attempted to go this route for method + field symbol, but it proved annoyingly complicated to get right in all cases)